### PR TITLE
kubectx: re-pack, switch from bash to go

### PR DESCRIPTION
**Pull request by jlesquembre**
###### Motivation for this change
With release 0.9, kubectx was rewritten in go. For more info see
https://github.com/ahmetb/kubectx/releases/tag/v0.9.0

As said in https://github.com/NixOS/nixpkgs/pull/86584#issuecomment-623306501 @periklis don't use this tool anymore, that's why I'm setting myself as maintainer. 
@periklis Is it ok with you? If not I'll update the PR to keep you as maintainer. 
If the reason to not use kubectx anymore is because you found an alternative, can you let me know which one? 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/kubectx/default.nix [hunk 1]
```diff
@@ -13,20 +11,15 @@ stdenv.mkDerivation rec {
     sha256 = "1b22jk8zl944w5zn3s7ybkkbmzp9519x32pfqwd1malfly7dzf55";
   };
 
-  buildInputs = [ makeWrapper ];
-
-  dontBuild = true;
-  doCheck = false;
+  goPackagePath = "github.com/ahmetb/kubectx";
```
Comment (jonringer) [inline pkgs/development/tools/kubectx/default.nix:14]: unused by buildGoModule
```suggestion
```

File: pkgs/development/tools/kubectx/default.nix [hunk 2]
```diff
@@ -13,20 +11,15 @@ stdenv.mkDerivation rec {
...
+  vendorSha256 = "168hfdc2rfwpz2ls607bz5vsm1aw4brhwm8hmbiq1n1l2dn2dj0y";
 
-  installPhase = ''
+  postInstall = ''
     mkdir -p $out/bin
     mkdir -p $out/share/zsh/site-functions
     mkdir -p $out/share/bash-completion/completions
     mkdir -p $out/share/fish/vendor_completions.d
```
Comment (jonringer) [inline pkgs/development/tools/kubectx/default.nix:21]: you can remove all the shell completion code, and replace it with:
```suggestion
  postInstall = ''
    installShellCompletion completion/*
  '';
```
still get:
```
$ tree ./result/share/
./result/share/
├── bash-completion
│   └── completions
│       ├── kubectx.bash
│       └── kubens.bash
├── fish
│   └── vendor_completions.d
│       ├── kubectx.fish
│       └── kubens.fish
└── zsh
    └── site-functions
        ├── _kubectx
        └── _kubens
```
```

File: pkgs/development/tools/kubectx/default.nix [hunk 3]
```diff
@@ -13,20 +11,15 @@ stdenv.mkDerivation rec {
     sha256 = "1b22jk8zl944w5zn3s7ybkkbmzp9519x32pfqwd1malfly7dzf55";
   };
 
-  buildInputs = [ makeWrapper ];
-
-  dontBuild = true;
-  doCheck = false;
+  goPackagePath = "github.com/ahmetb/kubectx";
+  vendorSha256 = "168hfdc2rfwpz2ls607bz5vsm1aw4brhwm8hmbiq1n1l2dn2dj0y";
```
Comment (jonringer) [inline pkgs/development/tools/kubectx/default.nix:16]: if you do the installShellCompleltion:
```suggestion
  nativeBuildInputs = [ installShellFiles ];
```

File: pkgs/development/tools/kubectx/default.nix [hunk 4]
```diff
@@ -1,8 +1,6 @@
-{ stdenv, lib, fetchFromGitHub, kubectl, makeWrapper }:
+{ stdenv, buildGoModule, fetchFromGitHub }:
```
Comment (jonringer) [inline pkgs/development/tools/kubectx/default.nix:1]:
```suggestion
{ stdenv, buildGoModule, fetchFromGitHub, installShellFiles }:
```

**Comment by jlesquembre** [comment]
@jonringer thanks for your review, it looks better now :)

**Comment by terinjokes** [comment]
Can this be reverted? The project states:

> *NOTE*: The *stable* versions of `kubectx` and `kubens` are the Bash script implementations (i.e. the `./kubectx` and `./kubens` files below). Go-based implementations are still *unstable* and are work-in-progress.

The Go ones are pretty unstable, such as not looking for kubeconfig files in the correct location (https://github.com/ahmetb/kubectx/issues/254) making this change unusable.

**Pull request merged**